tests/std/tests/Dev11_0836436_get_time/test.cpp

```diff
@@ -873,3 +875,33 @@ void test_gh_2618() {
     // 1-digit strings: same as 2-digit
     TestTimeGetYear("1", 2001, 1, 2001);
 }
+
+void test_gh_2848() {
+    //// GH-2848 <xloctime>: time_get::get can still assert 'istreambuf_iterator is not dereferenceable' when
+    ///the format is longer than the stream
+    {
+        const locale loc{locale::classic()};
+        const auto& tmget{use_facet<time_get<char>>(loc)};
+        ios_base::iostate err{ios_base::goodbit};
+        tm when{};
+        const string fmt{"%X"};
+        istringstream iss{"3:04"};
+        istreambuf_iterator<char> first{iss};
+        const istreambuf_iterator<char> last{};
+        tmget.get(first, last, iss, err, &when, fmt.data(), fmt.data() + fmt.size());
+        assert(err != ios_base::goodbit);

```

Comment:
More precisely, we want `err == ios_base::failbit | ios_base::eofbit` here (and below).